packages/generators/add-generator.ts

```diff
@@ -79,29 +84,39 @@ const searchProps = (answers, input) => {
  *
  */
 
-module.exports = class AddGenerator extends Generator {
+export default class AddGenerator extends Generator {
+	private dependencies: string[];
+	private configuration: {
+		config: {
+			configName?: string,
+			topScope?: string[],
+			item?: string;
+			webpackOptions?: any,
+		},
+	};
+
 	constructor(args, opts) {
 		super(args, opts);
 		this.dependencies = [];
 		this.configuration = {
 			config: {
+				topScope: ["const webpack = require('webpack')"],
 				webpackOptions: {},
-				topScope: ["const webpack = require('webpack')"]
-			}
+			},
 		};
 		const { registerPrompt } = this.env.adapter.promptModule;
 		registerPrompt("autocomplete", inquirerAutoComplete);
 	}
 
-	prompting() {
-		let done = this.async();
-		let action;
-		let self = this;
-		let manualOrListInput = action =>
-			Input("actionAnswer", `What do you want to add to ${action}?`);
+	public prompting() {
+		const done: any = this.async();
```

Comment:
This is a promise that resolves to a boolean if I remember right

Reply:
You mean something like?
```
const done: Promise<boolean>
```

Also, if it's a `done` is a promise, the compiler would fail all calls like `done()`. I think `done` is a function. Something like this should work:
```
const done: () => void | boolean
```

Reply:
It's a promise http://yeoman.io/authoring/running-context.html

Reply:
> If the asynchronous API you’re relying upon doesn’t support promises, then you can rely on the legacy this.async() way. Calling this.async() will return a function to call once the task is done.

It's not a promise. It could be something like this:
```js
const done: () => void | boolean
```